### PR TITLE
folderStructure.json refactor

### PR DESCRIPTION
**Pull request by Igorkowalski94**
#7911 

 If you encounter any issues or have any questions regarding the library, feel free to write [here](https://github.com/Igorkowalski94/eslint-plugin-project-structure/discussions/new?category=help), I’d be happy to help.

### CODE REVIEW

File: packages/twenty-front/folderStructure.json [hunk 1]
```diff
@@ -63,12 +51,12 @@
         { "name": "internal", "ruleId": "hooksLeafFolderRule" }
       ]
     },
+
     "doNotCheckLeafFolderRule": {
-      "folderRecursionLimit": 1,
       "children": [{ "name": "*" }, { "name": "*", "children": [] }]
```
Comment (Igorkowalski94) [inline packages/twenty-front/folderStructure.json:56]: It can be shortened to:
```suggestion
      "children": []
```

**Comment by Igorkowalski94** [comment]
@charlesBochet ready for review.

**Review by charlesBochet — APPROVED** [review]

**Comment by charlesBochet** [comment]
This is a nice improvement, thank you @Igorkowalski94

**Comment by Devessier** [comment]
Hello @Igorkowalski94,
I'm writing an article about how we use your great ESLint plugin at Twenty. I'm not sure if it's necessary to set the `regexParameters` to be able to assert camel and kebab cases. It seems that they are available by default. Am I missing something? Thank you so much.

```json
  "regexParameters": {
    "camelCase": "^[a-z]+([A-Za-z0-9]+)+",
    "kebab-case": "[a-z][a-z0-9]*(?:-[a-z0-9]+)*"
  },
```

**Comment by Igorkowalski94** [comment]
Hi @Devessier, thank you! 

Yes, they are available as default but with a different regex. @charlesBochet used different regexes for camelCase and kebab-case in the previous version, so I decided to keep them in regexParameters.

**Comment by Devessier** [comment]
Thank you a lot, @Igorkowalski94. I'm going to check that with Charles 😄

**Pull request merged**